addons/api/addon/serializers/host-catalog.js

```diff
@@ -4,30 +4,53 @@
  */
 
 import ApplicationSerializer from './application';
+import {
+  TYPE_HOST_CATALOG_PLUGIN_AWS,
+  TYPE_HOST_CATALOG_PLUGIN_AZURE,
+} from '../models/host-catalog';
 
-const fieldsByType = {
-  aws: [
-    'disable_credential_rotation',
-    'region',
+// These fields are Azure specific
+const azureFields = [
+  'disable_credential_rotation',
+  'tenant_id',
+  'client_id',
+  'subscription_id',
+  'secret_id',
+  'secret_value',
+];
+
+// These fields are AWS specific, AWS provider now has two credentialTypes: static and dynamic [not to be confused with static and dynamic host-catalogs]
+const AWSfieldsWithCredentialType = {
+  'static-credential': [
```

Comment:
Do we normally snake case these? I would have assumed this would be normal camel case

Reply:
I'm using the same var for translations and there we don't seem to use camel case, and for object keys we sometimes use snake case, but I'm open to changing it as I'm 100% convinced on my change

Reply:
Yeah I think we should be consistent here and stick with camel casing, our translations definitely will always be snake cased though

Reply:
update: we discussed this offline, since I'm using the same variable to dynamically generate translation and also in the model constants,  I'll stick with the snack case here.